Stack/Opc.Ua.Core/Stack/Server/ServerBase.cs

```diff
@@ -1152,7 +1152,7 @@ private static string GetBestDiscoveryUrl(Uri clientUrl, BaseAddress baseAddress
                         continue;
                     }
 
-                    if (endpointUrl.Port != baseAddress.Url.Port)
+                    if (endpointUrl.Port != baseAddress.Url.Port && !translateHttpsEndpoint)
```

Comment:
@BoBiene If this check is added than you will return duplicate HttpsBinaryTransport endpoints in case your configuration contains two such base addresses which differ only through their port numbers.

A better solution, which should fix also your case and the duplication issue that I previously mentioned on all profiles, is to completely remove this if clause and add a check at line 1180 

```
if (!translations.Exists(match => match.EndpointUrl == translation.EndpointUrl &&
                                 match.SecurityMode == translation.SecurityMode &&
                                 match.SecurityPolicyUri == translation.SecurityPolicyUri))
{
    translations.Add(translation);
}
```

Can you confirm this fixes your issue also ?

Reply:
Hi @mrsuciu,
I believe that the http endpoints with different ports should be specified separately. 
Since the ports are different, they are technically different endpoints.

Reply:
@BoBiene 
Indeed, but when such a configuration exists for the server, simply adding the code "&& !translateHttpsEndpoint" will have the secondary effect of duplicating the https endpoints returned to the client in the GetEndpoints service call.
You can confirm this by adding a second https endpoint under the <BaseAddresses> tag and check what the client receives when calling GetEndpoints.